### PR TITLE
Updating test instructions in TESTING.md

### PR DESCRIPTION
**Pull request by lukeatdell**
# Description
Updating test instructions in TESTING.md to clarify instructions and renaming unit-test make recipe to be more descriptive and consistent with other CSM repos.
- Renaming "gen-coverage" target to "unit-test".
- Updating github actions for running unit tests.
- Clarifying test execution instructions and adding instructions to test `repctl` in TESTING.md.
- Addressing areas that would require a user to have root or superuser permissions.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/1422 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Executing new make recipes
image
image
image

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -176,15 +176,15 @@ run-controller-tests:
...
-gen-coverage-csi-replicator:
+unit-test-csi-replicator:
 	go test ./controllers/csi-replicator/ ./pkg/connection/ -v -race -coverpkg=./controllers/csi-replicator/ -coverprofile cover.out
 
 #To generate coverage for the replication-controller
-gen-coverage-replication-controller:
+unit-test-replication-controller:
 	go test ./controllers/replication-controller/ -v -race -coverpkg=./controllers/replication-controller/ -coverprofile cover.out
 
 #To generate coverage for all the controllers
```
Comment (santhoshatdell) [inline Makefile:186]: Could you update the comment as well?

File: TESTING.md [hunk 2]
```diff
@@ -9,23 +10,48 @@ The integration test cases can be executed by using the `Makefile` recipe or fro
 		StorageClass: "replication-unity",
```
Comment (santhoshatdell) [inline TESTING.md:10]: Can we replace unity with one of supported platform?

File: Makefile (hunk 1 above)
Comment (lukeatdell) [reply Makefile]: done.

File: TESTING.md (hunk 2 above)
Comment (lukeatdell) [reply TESTING.md]: and done.

**Review by santhoshatdell — APPROVED** [review]

**Review by mbasha-dell — DISMISSED** [review]

**Review by santhoshatdell — APPROVED** [review]

**Review by mbasha-dell — APPROVED** [review]

**Pull request merged**